src/charm.py

```diff
@@ -28,142 +40,91 @@ def __init__(self, msg: str, status_type=None):
-                                "periodSeconds": 30,
-                            }
-                        },
-                    }
-                ],
-            },
-            {
-                "configMaps": {
-                    config["dashboard-configmap"]: {
-                        "settings": json.dumps(
-                            {
-                                "DASHBOARD_FORCE_IFRAME": True,
-                            }
-                        ),
-                        "links": Path("src/config.json").read_text(),
-                    },
-                },
-                "kubernetesResources": {
-                    "customResources": {
-                        "profiles.kubeflow.org": [
-                            {
-                                "apiVersion": "kubeflow.org/v1beta1",
-                                "kind": "Profile",
-                                "metadata": {"name": config["profile"]},
-                                "spec": {
-                                    "owner": {"kind": "User", "name": config["profile"]}
-                                },
-                            }
-                        ]
+    def get_kubeflow_dashboard_operator_layer(self, profiles_service: str) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+        layer_config = {
+            "summary": "dex-auth-operator layer",
+            "description": "pebble config layer for kubeflow_dashboard_operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the kubeflow_dashboard_operator image",
+                    "command": f"{self._entrypoint}",
+                    "startup": "enabled",
+                    "environment": {
+                        "USERID_HEADER": "kubeflow-userid",
+                        "USERID_PREFIX": "",
+                        "PROFILES_KFAM_SERVICE_HOST": f"{profiles_service}.{self.model.name}",
+                        "REGISTRATION_FLOW": self.model.config["registration-flow"],
+                        "DASHBOARD_LINKS_CONFIGMAP": self.model.config[
+                            "dashboard-configmap"
+                        ],
                     },
-                },
+                }
             },
-        )
-
-        self.model.unit.status = ActiveStatus()
+        }
+        return Layer(layer_config)
 
-    def _configure_mesh(self, interfaces):
-        if interfaces["ingress"]:
-            interfaces["ingress"].send_data(
-                {
-                    "prefix": "/",
-                    "rewrite": "/",
-                    "service": self.model.app.name,
-                    "port": self.model.config["port"],
-                }
-            )
+    def _check_container_connection(self):
+        if not self._container.can_connect():
+            raise CheckFailed("Waiting for pod startup to complete", WaitingStatus)
```

Comment:
Is not there a chcance that after a while this will be ok ? I am not sure here if I get the container states correctly :(

Reply:
Mmmh, yes this could eventually be okay because you are binding all hooks to the same event handler (`main`), though exceptions you are not handling will be raised if you happen to run into the first line for `_update_layer`, which may cause unexpected behaviours.

Reply:
make sense thanks :)